### ISSUE TITLE
Use GitHub API to provide more info about Repo

### ISSUE THREAD

**Issue by waylan** [mkdocs/mkdocs#504]
I expect this is probably a Post-1.0 feature. However, I want to record my thoughts before I forget.

Currently, the settings config includes a setting to point to a GitHub or Bitbucket repo and then allows a link to be included in the template. However, it might be nice to be able to include other data in the template as well. In fact, GitHub provides an [API][1] which could be used to fetch all sorts of data about a repo (I'm not familiar with what Bitbucket offers, if anything, so I'm focusing on GitHub for now).

Interestingly, while I've been thinking about this for a few weeks now, just yesterday GitHub [announced][2] that they are expanding the repo info then provide in [GitHub Pages][3] to include [releases][4] meta-data. The announcement points to the [docs][5] for all the repo related data available within GitHub Pages. I'm not sure mkdocs should supply all that data, however, a link to the latest release would be very useful, among a few others. As the announcement states (in part), ...

> allowing you to more easily display information about your project's latest version (including release notes) and link directly to download the most recent releases.

It occurs to me that this might be a good candidate for something that should be implemented using a (future) Plugin API (see #206) -- especially given that GitHub may be different than other repo hosts (like Bitbucket). Whether it is built into mkdocs proper or as part of a plugin system, I would love to see `build` make a call to the GitHub API (based on the repo set in the config)  and then add a bunch of repo meta-data to the template context. Of course, this would require network access during a build and `serve` should probably cache the data to avoid a request on every reload.

[1]: https://developer.github.com/v3/repos/
[2]: https://github.com/blog/1996-releases-metadata-for-github-pages
[3]: https://help.github.com/articles/what-are-github-pages
[4]: https://github.com/blog/1547-release-your-software
[5]: https://help.github.com/articles/repository-metadata-on-github-pages/

**Comment by d0ugal** [comment]
> It occurs to me that this might be a good candidate for something that should be implemented using a (future) Plugin API

Yup, that was my thought here. It would be interesting to see what this would look like exactly. We already do some GitHub specific stuff, so in some ways it is a natural progression.

**Comment by timvink** [comment]
If you're looking to extract information from github (like your github avatar and such), have a look at https://github.com/ojacques/mkdocs-git-committers-plugin-2 also!

More background info on adding git info (to mkdocs material theme): https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/?h=git
